kernelci/cli/storage.py

```diff
@@ -0,0 +1,38 @@
+# SPDX-License-Identifier: LGPL-2.1-or-later
+#
+# Copyright (C) 2023 Collabora Limited
+# Author: Guillaume Tucker <guillaume.tucker@collabora.com>
+
+"""Tool to interact with storage services supported by KernelCI"""
+
+import os
+
+import click
+
+import kernelci.config
+import kernelci.storage
+from . import Args, kci
+
+
+@kci.group(name='storage')
+def kci_storage():
+    """Interact with KernelCI storage services"""
+
+
+@kci_storage.command(secrets=True)
+@click.argument('filename', type=click.Path(exists=True))
+@click.argument('path', required=False)
+@Args.config
+@Args.storage
+def upload(filename, path, config, storage, secrets):
+    """Upload FILENAME to the designated storage service in PATH"""
+    configs = kernelci.config.load(config)
+    storage_config = configs['storage'][storage]
+    storage = kernelci.storage.get_storage(
+        storage_config, secrets.storage.credentials
+    )
+    url = storage.upload_single(
+        (filename, os.path.basename(filename)),
+        path or ''
```

Comment:
Well is this really necessary?  I'm not sure the named arguments provide much improvement here.

The whole `Storage.upload_single()` thing is a bit convoluted to be honest, I think that can be made clearer and then calling it would become clearer too.

Reply:
Yes and that's outside the scope of this PR.  Like I wrote, the Storage class needs to be improved.

I really don't think this suggestion is adding any clarity as what's the semantics of `file_path` and `dest_path`?  But I can add that anyway if you want.